src/pages/form/[id].tsx

```diff
@@ -24,13 +35,71 @@ const Form = () => {
       uiSchema: JSON.parse(data.form.uiSchema),
     };
   }, [data]);
-  if (isLoading) return <DefaultLoadingPage />;
-  if (error) return <ErrorPage errorMessage={error.message} />;
+  const user = useSession();
+
+  const wasSubmitted = useMemo(() => {
+    const wasSubmitted = submssionData?.find(
+      (submission) => submission.submittedById === user?.data?.user?.id,
+    );
+    return wasSubmitted;
+  }, [submssionData, user?.data?.user?.id]);
+  const countdownHome = () => {
+    if (countdown <= 0) {
+      router
+        .push("/")
+        .then()
+        .catch((error) => console.error(error));
+    } else {
+      setCountdown(countdown - 1);
+    }
+  };
+  if ((submitted || wasSubmitted) ?? user.status === "unauthenticated") {
```

Comment:
This will create a timeout every time this component rerenders, put this in a useEffect or somewhere else